### ISSUE TITLE
Be able to push first on a Git repository

### ISSUE THREAD

**Issue by dinosaure** [mirage/ocaml-git#453]
Currently, if we want to push to an empty Git repository. we got an error. We should be able to push first. I suspect that the push command expects at least some references.

**Comment by hannesm** [comment]
note: in git 2.x this used to work :)

**Comment by hannesm** [comment]
I'm still seeing this issue. My setup is as follows:
on the host side, I initialize an empty git repository:
```bash
cd /usr/local/git/test.git && rm -rf * && git init --bare && touch git-daemon-export-ok && chown -R git_daemon .
```

I start a git daemon (with `--enable=receive-pack --base-path=/usr/local/git`).

On the unikernel / ocaml-git side, I have the following:
```OCaml
  module Git_store = Irmin_mirage_git.KV_RW(Irmin_git.Mem)(Pclock)

  let start ... =
    Irmin_git.Mem.v (Fpath.v "somewhere") >>= function
    | Error _ -> invalid_arg "Could not create an in-memory git repository."
    | Ok git ->
      let author _ = "keyfender"
      and msg _ = "a keyfender change"
      in
      Git_store.connect git ~depth:1 ~ctx ~author ~msg (Key_gen.remote ()) >>= fun store ->
      Git_store.set store (Fpath.v ".gitignore") "" >>= function
        | Ok () -> Lwt.return_unit
        | Error e ->
          Logs.err (fun m -> m "couldn't write to store %a" Git_store.pp_write_error e);
          Lwt.fail_with "store not writable")
```

I'm reachting the Error case, log output:
```
2021-02-01 11:41:50 -00:00: DBG [smart_flow] Write 59 byte(s).
2021-02-01 11:41:50 -00:00: DBG [smart_flow] Start to read 65536 byte(s).
2021-02-01 11:41:50 -00:00: DBG [smart_flow] Got 141/65536 byte(s).
2021-02-01 11:41:50 -00:00: ERR [smart_flow] Got a protocol error: Invalid PKT-line.
2021-02-01 11:41:50 -00:00: ERR couldn't write to store Invalid PKT-line
Fatal error: exception (Failure "store not writable")
```

The wireshark "follow TCP flow" shows the following:
```
CLIENT: 0032git-receive-pack /data.git.host=192.168.23.1.
SERVER: 008d0000000000000000000000000000000000000000 capabilities^{}.report-status delete-refs side-band-64k quiet atomic ofs-delta agent=git/2.27.0
SERVER: 0000
```

For comparison, a session with cgit on the client side:
```
CLIENT: 002Egit-receive-pack /data.git.host=127.0.0.1.
SERVER: 008d0000000000000000000000000000000000000000 capabilities^{}.report-status delete-refs side-band-64k quiet atomic ofs-delta agent=git/2.27.0
0000
CLIENT: 00950000000000000000000000000000000000000000 1abce0c558cc5909581bb8a3ae9eb9e4f61b4802 refs/heads/master. report-status side-band-64k agent=git/2.27.00000PACK.........
x...1..!....Wlor..;.M.......X.	...D.`9....`..}[..Xr..$t.e...Y..
......1..{...R.L.s...~^.|^.<..w....h.h..*.v...254...x.340031Q.K.,.L../Jex6...M..9{wk.+..q.IO.D....N0x........T..i6szA:7.D.B.2b.W
SERVER: 0030.000eunpack ok
0019ok refs/heads/master
00000000
```

**Comment by hannesm** [comment]
this is fixed with #456 :D thanks

**Comment by dinosaure** [comment]
Finally fixed.
